common/djangoapps/util/enterprise_helpers.py

```diff
@@ -70,6 +72,75 @@ def post_enterprise_course_enrollment(self, username, course_id, consent_granted
             raise EnterpriseApiException(message)
 
 
+def data_sharing_consent_required(return_to=None):
+    """
+    Decorator which makes a view method redirect to the Data Sharing Consent form if:
+
+    * The wrapped method is passed request, course_id as the first two arguments.
+    * Enterprise integration is enabled
+    * Data sharing consent is required before accessing this course view.
+    * THe request.user has not yet given data sharing consent for this course.
+
+    Usage:
+
+    * Pass the name of the view to redirect to after consent is given:
+
+        @data_sharing_consent_required('courseware')
+        def my_view(request, course_id, ...):
+            # Can now assume that if course_id requires data sharing consent, that request.user has granted it.
+            # ...
+
+    * OR, after consent is given, redirect to 'my_view'
+
+        @data_sharing_consent_required()
+        def my_view(request, course_id, ...):
+            # If consent is required, the user is redirected to the URL named my_view
+            # ...
+
+    * Warning: The parentheses are required when using this decorator.  This will _not_ work:
+
+        @data_sharing_consent_required
+        def my_view(request, course_id, ...):
+            # Throws an error when my_view is called.
+
+    Arguments:
+    * return_to: url name label for the page to return to once consent is granted.
+        Defaults to the name of the wrapped function.
```

Comment:
Oh wow, removing that parameter makes things so much cleaner, and would have saved me a lot of pain!  I had been looking at `@login_required`, but apparently not closely enough.  Thanks for your advice!

Fixed with c497322.